pkg/mynewt-core/include/os/os.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_mynewt_core
+ * @{
+ *
+ * @file
+ * @brief       mynewt-core ) error types
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
```

Comment:
not sure, but this file is mostly copied over from "MYNEWT/kernel/os/include/os/os.h", is it not? Could this bring is into trouble regarding licence issues or just marking it as one of our own files?

Reply:
This might apply for more files in this include folder, right?!

Reply:
True... I'll see if I can figure out a less invasive solution...

Reply:
I'll change the license header in most cases and add a change notice when it applies. But in many cases I don't think it applies since its just interface definitions.

For some header definitions, I could simply patch the ones in mynewt-core as well:

```
kernel/os/include/os/os_callout.h
kernel/os/include/os/os_mutex.h
kernel/os/include/os/os_sem.h
kernel/os/include/os/os_task.h
```

But since you prefer to avoid patches should I keep them here?

Reply:
I addressed some of them, let me know what you prefer for the others.

Reply:
https://github.com/RIOT-OS/RIOT/pull/16348/commits/6dc1478aff611c6617746a4bde395f41b813d3bd implements that idea, but I can revert it.

Reply:
In that case, I think patching would be the safer bet. I would rather take this (potentially wide) detour around potential license issues than avoid patches in any case here. Better safe than sorry :-)